packages/plugin/src/index.js

```diff
@@ -1,17 +1,48 @@
-import React from 'react'
+import React, { useEffect, useState } from 'react'
 import { VIS_TYPE_PIVOT_TABLE } from '@dhis2/analytics'
+import { useDataEngine } from '@dhis2/app-runtime'
 
+import { apiFetchLegendSet } from './api/legendSets'
 import ChartPlugin from './ChartPlugin'
 import PivotPlugin from './PivotPlugin'
 
 const VisualizationPlugin = props => {
+    const engine = useDataEngine()
+    const [legendSet, setLegendSet] = useState(null)
+
+    const hasLegendSet =
+        props.visualization.legendSet && props.visualization.legendSet.id
+
+    useEffect(() => {
+        const fetchLegendSet = async engine => {
+            if (hasLegendSet) {
+                const response = await apiFetchLegendSet(
+                    engine,
+                    props.visualization.legendSet.id
+                )
+
+                if (response && response.legendSet) {
+                    setLegendSet(response.legendSet)
+                }
+            }
+        }
+
+        fetchLegendSet(engine)
+    }, [props.visualization.legendSet]) // eslint-disable-line react-hooks/exhaustive-deps
```

Comment:
I don't think we need to disable `eslint` here right?  The only dependencies are `props.visualization.legendSet`, `engine` (which we should add since it's constant), and maybe `hasLegendSet`.  Is there a reason not to list them all?

Reply:
No, I don't see any reason.
The eslint rule is probably a leftover from the start.